rust/src/interpreter/memory.rs

```diff
@@ -0,0 +1,354 @@
+use std::{cmp::max, iter};
+
+use evmc_vm::StatusCode;
+
+use crate::{
+    interpreter::{
+        gas::{consume_copy_cost, consume_gas},
+        utils::word_size,
+    },
+    types::u256,
+};
+
+#[derive(Debug)]
+pub struct Memory(Vec<u8>);
+
+impl Memory {
+    pub fn new(memory: Vec<u8>) -> Self {
+        Self(memory)
+    }
+
+    pub fn into_inner(self) -> Vec<u8> {
```

Comment:
Is this ever used?  
Shall we provide &[u8] getter and let the caller side decide what to do with it?

Reply:
Yes, the vector is needed in the ffi module. By providing only `&[u8]` a new allocation would be needed.

Reply:
OK, I still do not like the name. Into_inner is not a good name, leaks implementation.

Reply:
I kind of agree but on the flip side `.into_inner()` is not uncommon. e.g. `Box::into_inner` or `BufReader::into_inner`. And the fact that `Memory` holds a `Vec` could be also inferred from `Memory::new` which takes a `Vec`. But I would also be fine with renaming it.